test/IntegrationTests/test_getallchats.py

```diff
@@ -0,0 +1,317 @@
+
+        # Verify that the chat_id and customer_guid in the response match the input
+        if messages:  # Check if messages list is not empty
+            # Assuming the first message corresponds to the chat_id and customer_guid
+            first_message = messages[0]
+            self.assertEqual(first_message['chat_id'], self.valid_chat_id,
+                             "Chat ID in response does not match the input chat ID")
+            self.assertEqual(first_message['customer_guid'], self.valid_customer_guid,
+                             "Customer GUID in response does not match the input customer GUID")
+
+        # Check for reverse chronological order
+        for i in range(len(messages) - 1):
+            current_timestamp = messages[i]['timestamp']
+            next_timestamp = messages[i + 1]['timestamp']
+            self.assertGreaterEqual(current_timestamp, next_timestamp,
+                                    "Messages are not in reverse chronological order")
+
+        logger.info("=== Test Case 4 Completed ===\n")
+
+    def test_retrieval_with_page_two(self):
+        """Test case 5: Valid customer_guid, chat_id, page=2, and page_size=10"""
+        logger.info("=== Starting Test Case 5: Valid customer_guid, chat_id, page=2, and page_size=10 ===")
+
+        # Assuming we have enough messages to test pagination
+        for page in range(1, 4):  # Test first three pages
+            url = f"{self.BASE_URL}/getallchats"
+            data = {
+                "customer_guid": self.valid_customer_guid,
+                "chat_id": self.valid_chat_id,
+                "page": page,
+                "page_size": 5
+            }
+            logger.info(f"INPUT: Sending request with data:\n{str(data)}")
+
+            response = requests.post(url, json=data)
+            logger.info(f"OUTPUT: Response status code: {response.status_code}")
+            logger.info(f"OUTPUT: Response content: {response.text}")
+
+            if response.status_code == 200:
+                messages = response.json().get('messages')
+                self.assertIsInstance(messages, list)
+                self.assertLessEqual(len(messages), 5, "Messages returned exceed page size")
+            else:
+                self.assertEqual(response.status_code, 404, "Expected 404 for no more messages")
+
+            logger.info("=== Test Case 5 Completed ===\n")
+
+    def test_single_message_retrieval(self):
+        """Test case 6: Valid customer_guid, chat_id, page=1, and page_size=1"""
+        logger.info("=== Starting Test Case 6: Valid customer_guid, chat_id, page=1, and page_size=1 ===")
+
+        url = f"{self.BASE_URL}/getallchats"
+        data = {
+            "customer_guid": self.valid_customer_guid,
+            "chat_id": self.valid_chat_id,
+            "page": 1,
+            "page_size": 1
+        }
+        logger.info(f"INPUT: Sending request with data:\n{str(data)}")
```

Comment:
First we should add a chat message and then verify retrieving it with page as 1 and page_size as 1 and verify the one retrieved should exactly match the input

Reply:
Done sir, chat message added and verified with exact match.